packages/shylu/shylu_dd/frosch/src/Adapters/Thyra_FROSchFactory_def.hpp

```diff
@@ -108,18 +108,17 @@ namespace Thyra {
         TEUCHOS_TEST_FOR_EXCEPT((bIsEpetra != bIsTpetra) && bIsBlocked == true);
         
         // Retrieve Matrix
-        RCP<Matrix<SC,LO,GO,NO> > A = Teuchos::null;
         RCP<const CrsMatrix<SC,LO,GO,NO> > xpetraFwdCrsMat = ThyraUtils<SC,LO,GO,NO>::toXpetra(fwdOp);
         TEUCHOS_TEST_FOR_EXCEPT(is_null(xpetraFwdCrsMat));
         
-        // FROSCH needs a non-const object as input
+        // AH 08/07/2019: Going from const to non-const to const. One should be able to improve this.
         RCP<CrsMatrix<SC,LO,GO,NO> > xpetraFwdCrsMatNonConst = rcp_const_cast<CrsMatrix<SC,LO,GO,NO> >(xpetraFwdCrsMat);
         TEUCHOS_TEST_FOR_EXCEPT(is_null(xpetraFwdCrsMatNonConst));
```

Comment:
This can't be null, since `rcp_const_cast` won't fail at run time.

Reply:
Thanks, I removed this as well.